### ISSUE TITLE
"Could not find manifest file" during option 6 / fpga-gnw-romgenerator

### ISSUE THREAD

**Issue by Tolaris** [mattpannella/pocket-updater-utility#164]
Using Analogue Pocket Core Updater v2.30.0, and following the published instructions to place artwork and roms in Assets/gameandwatch/agg23.GameAndWatch/{artwork, roms}, I get this error when running on Linux:

```$ ./pocket_updater_x64
Analogue Pocket Core Updater v2.30.0
Checking for updates...
...

Choose your destiny: 6
Executing /home/tjwagner/Private/Games/Pocket/tools/gameandwatch/linux/fpga-gnw-romgenerator
thread 'main' panicked at 'Could not find manifest file: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/main.rs:118:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Adding the RUST_BACKTRACE args:

```$ RUST_BACKTRACE=1 ./pocket_updater_x64
Analogue Pocket Core Updater v2.30.0
Checking for updates...
...

Choose your destiny: 6
Executing /home/tjwagner/Private/Games/Pocket/tools/gameandwatch/linux/fpga-gnw-romgenerator
thread 'main' panicked at 'Could not find manifest file: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/main.rs:118:54
stack backtrace:
   0: rust_begin_unwind
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:579:5
   1: core::panicking::panic_fmt
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:64:14
   2: core::result::unwrap_failed
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/result.rs:1750:5
   3: fpga_gnw_romgenerator::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


$ RUST_BACKTRACE=full ./pocket_updater_x64
Analogue Pocket Core Updater v2.30.0
Checking for updates...
...

Choose your destiny: 6
Executing /home/tjwagner/Private/Games/Pocket/tools/gameandwatch/linux/fpga-gnw-romgenerator
thread 'main' panicked at 'Could not find manifest file: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/main.rs:118:54
stack backtrace:
   0:     0x558e68e4b25a - std::backtrace_rs::backtrace::libunwind::trace::ha9053a9a07ca49cb
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x558e68e4b25a - std::backtrace_rs::backtrace::trace_unsynchronized::h9c2852a457ad564e
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x558e68e4b25a - std::sys_common::backtrace::_print_fmt::h457936fbfaa0070f
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x558e68e4b25a - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h5779d7bf7f70cb0c
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x558e68e6c87e - core::fmt::write::h5a4baaff1bcd3eb5
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/fmt/mod.rs:1232:17
   5:     0x558e68e48655 - std::io::Write::write_fmt::h4bc1f301cb9e9cce
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/io/mod.rs:1684:15
   6:     0x558e68e4b025 - std::sys_common::backtrace::_print::h5fcdc36060f177e8
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x558e68e4b025 - std::sys_common::backtrace::print::h54ca9458b876c8bf
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x558e68e4c7df - std::panicking::default_hook::{{closure}}::hbe471161c7664ed6
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:271:22
   9:     0x558e68e4c51b - std::panicking::default_hook::ha3500da57aa4ac4f
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:290:9
  10:     0x558e68e4cd88 - std::panicking::rust_panic_with_hook::h50c09d000dc561d2
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:692:13
  11:     0x558e68e4cc89 - std::panicking::begin_panic_handler::{{closure}}::h9e2b2176e00e0d9c
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:583:13
  12:     0x558e68e4b6c6 - std::sys_common::backtrace::__rust_end_short_backtrace::h5739b8e512c09d02
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:150:18
  13:     0x558e68e4c992 - rust_begin_unwind
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:579:5
  14:     0x558e68c3a383 - core::panicking::panic_fmt::hf33a1475b4dc5c3e
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:64:14
  15:     0x558e68c3a833 - core::result::unwrap_failed::hdff5465d74574b44
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/result.rs:1750:5
  16:     0x558e68c67fe9 - fpga_gnw_romgenerator::main::h140efe405af182a2
  17:     0x558e68c8c5f3 - std::sys_common::backtrace::__rust_begin_short_backtrace::ha1f03f2a0b65eeaf
  18:     0x558e68c8c7a9 - std::rt::lang_start::{{closure}}::hae2ddcbad66e9ac8
  19:     0x558e68e42b4c - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hd6efcd3bec896f2c
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:287:13
  20:     0x558e68e42b4c - std::panicking::try::do_call::hce04e543bb1f4cbb
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:487:40
  21:     0x558e68e42b4c - std::panicking::try::h3342dd4e1f680968
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:451:19
  22:     0x558e68e42b4c - std::panic::catch_unwind::h148ce1e59ac0cee7
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panic.rs:140:14
  23:     0x558e68e42b4c - std::rt::lang_start_internal::{{closure}}::h25f9dda2057a67fe
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/rt.rs:148:48
  24:     0x558e68e42b4c - std::panicking::try::do_call::h7caaaeaf9401650b
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:487:40
  25:     0x558e68e42b4c - std::panicking::try::he7d15285746cbbc2
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:451:19
  26:     0x558e68e42b4c - std::panic::catch_unwind::h89fb4f50c0301fe0
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panic.rs:140:14
  27:     0x558e68e42b4c - std::rt::lang_start_internal::h078acd489417d3c1
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/rt.rs:148:20
  28:     0x558e68c6ea85 - main
  29:     0x7efd5863b18a - <unknown>
  30:     0x7efd5863b245 - __libc_start_main
  31:     0x558e68c3a9ca - _start
                               at /Volumes/x64-linux-build/.build/src/glibc-2.26/csu/../sysdeps/x86_64/start.S:120
  32:                0x0 - <unknown>
```

It looks like redownloading the tools works: https://www.reddit.com/r/AnaloguePocket/comments/14jx17b/game_watch_core_released/

Perhaps you need to repackage the tools?

**Comment by Tolaris** [comment]
Ah, the version of agg23.GameAndWatch_0.1.0_2023-06-26-Tools.zip is the same and I found manifest.json is there:

```
$ find tools/gameandwatch/
tools/gameandwatch/
tools/gameandwatch/windows
tools/gameandwatch/windows/fpga-gnw-romgenerator.exe
tools/gameandwatch/windows/manifest.json
tools/gameandwatch/agg23.GameAndWatch_0.1.0_2023-06-26-Tools.zip
tools/gameandwatch/mac
tools/gameandwatch/mac/fpga-gnw-romgenerator
tools/gameandwatch/mac/manifest.json
tools/gameandwatch/linux
tools/gameandwatch/linux/fpga-gnw-romgenerator
tools/gameandwatch/linux/manifest.json
```

Looks like the issue is just in invocation.

**Comment by Tolaris** [comment]
Left some comments on https://github.com/mattpannella/pocket-updater-utility/commit/9d14b021cea2d59f671b66ca4d8583c7da1953f0 which I think addresses this.

**Comment by Tolaris** [comment]
Manual execution works correctly. Issue is just the way we're building the manifest_path arg:

$ ./tools/gameandwatch/linux/fpga-gnw-romgenerator --mame-path Assets/gameandwatch/agg23.GameAndWatch --output-path Assets/gameandwatch/common --manifest-path ./tools/gameandwatch/linux/manifest.json supported
-------------------------
Processing device gnw_ball
...

**Comment by mattpannella** [comment]
wooops. thanks haha
I threw this together last night while I was tired and didn't feel like testing it on anything but my MacBook
I'll get a patch out in a bit

**Comment by mattpannella** [comment]
should be fix by #165
